[0.6.x] ScalaJSDefined.lazy_vals fails on Scala 2.12.0 (only)

Discovered by the weekly build:
```
[info] Test org.scalajs.testsuite.jsinterop.ScalaJSDefinedTest.lazy_vals started
[error] Test org.scalajs.testsuite.jsinterop.ScalaJSDefinedTest.lazy_vals failed: java.lang.AssertionError: expected:<42> but was:<undefined>, took 0.000999936 sec
[error]     at org.scalajs.testsuite.jsinterop.ScalaJSDefinedTest.lazy$undvals(...)
[error]     at org.scalajs.testsuite.jsinterop.ScalaJSDefinedTest$scalajs$junit$bootstrapper$.invoke(...)
[error]     ... 3 more
```
This issue is 0.6.x-only simply because Scala.js 1.x dropped support for 2.12.0.

Needs investigation, but I'm betting that the blame will be on scalac upstream.

--- a/compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala
+++ b/compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala
@@ -5605,10 +5605,14 @@ abstract class GenJSCode extends plugins.PluginComponent
    */
   private def isExposed(sym: Symbol): Boolean = {
     !sym.isBridge && {
-      if (sym.isLazy)
-        sym.isAccessor && sym.accessed.hasAnnotation(ExposedJSMemberAnnot)
-      else
+      if (sym.isLazy) {
+        sym.isAccessor && {
+          sym.accessed.hasAnnotation(ExposedJSMemberAnnot) ||
+          sym.hasAnnotation(ExposedJSMemberAnnot) // for 2.12.0, see #3439
+        }
+      } else {
         sym.hasAnnotation(ExposedJSMemberAnnot)
+      }
     }
   }
 

